recipes/autoconf/all/conanfile.py

```diff
@@ -13,110 +18,77 @@ class AutoconfConan(ConanFile):
-            autotools = self._configure_autotools()
-            autotools.make()
+        apply_conandata_patches(self)
+        self._autotools()
 
     def package(self):
-        self.copy("COPYING*", src=self._source_subfolder, dst="licenses")
-        with self._build_context():
-            autotools = self._configure_autotools()
-            autotools.install()
-        tools.rmdir(os.path.join(self.package_folder, "bin", "share", "info"))
-        tools.rmdir(os.path.join(self.package_folder, "bin", "share", "man"))
+        autotools = self._autotools()
+        # KB-H013 we're packaging an application, place everything under bin
+        autotools.install(args=[f"DESTDIR={unix_path(self, path.join(self.package_folder, 'bin'))}"])
+        copy(self, "COPYING*", src=self.source_folder, dst=path.join(self.package_folder, "licenses"))
+        rmdir(self, path.join(self._datarootdir, "info"))
+        rmdir(self, path.join(self._datarootdir, "man"))
+
+    def _set_env(self, var_name, var_path):
+        self.output.info(f"Setting {var_name} to {var_path}")
+        self.buildenv_info.define_path(var_name, var_path)
+        setattr(self.env_info, var_name, var_path)
 
     def package_info(self):
-        self.cpp_info.libdirs = []
-        self.cpp_info.includedirs = []
-
-        bin_path = os.path.join(self.package_folder, "bin")
-        self.output.info("Appending PATH env var with : {}".format(bin_path))
-        self.env_info.PATH.append(bin_path)
-
-        ac_macrodir = self._autoconf_datarootdir
-        self.output.info("Setting AC_MACRODIR to {}".format(ac_macrodir))
-        self.env_info.AC_MACRODIR = ac_macrodir
-
-        autoconf = tools.unix_path(os.path.join(self.package_folder, "bin", "autoconf"))
-        self.output.info("Setting AUTOCONF to {}".format(autoconf))
-        self.env_info.AUTOCONF = autoconf
-
-        autoreconf = tools.unix_path(os.path.join(self.package_folder, "bin", "autoreconf"))
-        self.output.info("Setting AUTORECONF to {}".format(autoreconf))
-        self.env_info.AUTORECONF = autoreconf
-
-        autoheader = tools.unix_path(os.path.join(self.package_folder, "bin", "autoheader"))
-        self.output.info("Setting AUTOHEADER to {}".format(autoheader))
-        self.env_info.AUTOHEADER = autoheader
-
-        autom4te = tools.unix_path(os.path.join(self.package_folder, "bin", "autom4te"))
-        self.output.info("Setting AUTOM4TE to {}".format(autom4te))
-        self.env_info.AUTOM4TE = autom4te
-
-        autom4te_perllibdir = self._autoconf_datarootdir
-        self.output.info("Setting AUTOM4TE_PERLLIBDIR to {}".format(autom4te_perllibdir))
-        self.env_info.AUTOM4TE_PERLLIBDIR = autom4te_perllibdir
+        # KB-H013 we're packaging an application, hence the nested bin
+        bin_dir = path.join(self.package_folder, "bin", "bin")
+        self.output.info(f"Appending PATH env var with : {bin_dir}")
+        self.buildenv_info.prepend_path("PATH", bin_dir)
```

Comment:
```suggestion
```

Did I not review this part already? It's useless.

Reply:
Yes you did review that part, but that was on account of the everything placed in the layout folders. Since this recipe is a build tool all the folders are placed in the bin folder.
So basically the folder containing the actual binaries is `self.package_folder/bin/bin` since the layout bindirs only add the `self.package_folder/bin` to the path, I still needed to add this embedded bin dit to the buildenv_info.
I tried to solve this with specifying the layout structure, while adhering to kb-h013 but that caused more problems then simply adding this line